### PR TITLE
ci: Add Rusk build workflow

### LINKED ISSUE #3143: ci: Add binary build workflow for Rusk
#### Summary
To streamline participation and automate versioning, we need to offer pre-built binaries for the different Rusk node roles: Provisioner, archive node and prover.

These binaries should be platform-specific and versioned for compatibility with network releases.

#### Possible solution design or implementation
The easiest way to do this is to create a Actions workflow similar to what we have for `Rusk wallet`. We should have a build matrix with relevant OSs and architecture targets. 

We should make artifacts for the different node roles as well.

### PR DESCRIPTION
**Pull request by HDauven**
Resolves #3143

### CODE REVIEW

File: .github/workflows/rusk_build.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  build_and_publish:
+    name: Build Rusk binaries for ${{ matrix.os }} (${{ matrix.features }})
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-24.04, macos-12, arm-linux]
+        compiler: [cargo]
+        features: [default, archive]
+        include:
+          - os: ubuntu-24.04
```
Comment (herr-seppia) [inline .github/workflows/rusk_build.yml:21]: Are we moving support from 22.04 to 24.04?

Comment (HDauven) [reply .github/workflows/rusk_build.yml:21]: @herr-seppia we don't explicitly mention 22.04 or 24.04 anywhere in our Rusk repo. I've been testing the binaries on 22.04, 24.04 and 24.10, and they all work fine on my end.

In my opinion it makes sense to explicitly state we support 24.04, given it has been out for 8 months now and "End of Standard Support" runs to June 2029. But I don't have a very strong opinion here. What do you think?

**Comment by HDauven** [comment]
@herr-seppia This currently only builds the binaries. What do you think about key creation and contract compilation? Should we keep it outside of this workflow and rely on the methods to retrieve them we currently use in the `node-installer`?

**Comment by Neotamandua** [comment]
> @herr-seppia This currently only builds the binaries. What do you think about key creation and contract compilation? Should we keep it outside of this workflow and rely on the methods to retrieve them we currently use in the `node-installer`?

I like the idea of pushing this down to Rusk. This also allows for more use cases, for example adding another workflow in the future related to https://github.com/dusk-network/rusk/issues/2184 when contracts are compiled.

**Comment by HDauven** [comment]
> What would be the benefit of having contracts and key built on CI?
> 
> Currently, the keys are self built by the rusk binary, and the contracts are embedded in the genesis state

You're right, I'll leave it out

**Review by Neotamandua — APPROVED** [review]

**Pull request merged**